sourcecode/weather_reports_display.py

```diff
@@ -0,0 +1,205 @@
+import calendar
+from datetime import datetime
+
+from .weather_records_calculation import WeatherRecordsCalculation
+from .weather_records_reader import WeatherRecordsReader
+
+BLUE_COLOR = "\033[;34;40m"
+RED_COLOR = "\033[;31;40m"
+WHITE_COLOR = "\033[;;40m"
+
+class WeatherReportsDisplay():
+    def __init__(self, path: str) -> None:
+        """
+        Constructor for initializing reports
+        
+        @params
+        :path str: Complete path of weatherfiles 
+        """
+        self.path = path
+        self.year_report = {
+            'highest_temp': float('-inf'),
+            'highest_temp_date': "",
+            'lowest_temp_date': "",
+            'lowest_temp': float('inf'),
+            'humidity': float('-inf'),
+            'humidity_date': "",
+        }
+
+        self.month_report = {
+            'avg_highest_temp': float('-inf'),
+            'avg_lowest_temp': float('inf'),
+            'avg_mean_humidity': float('-inf'),
+        }
+
+    def display_extreme_report_of_year(self, date: str) -> None:
+        """
+        Used for printing report
+        
+        @params        
+        :path str: Contains path to weather files directory.
+        :date str: Date entered by user as command line argument.
+        """
+        weather_files = WeatherRecordsReader.weather_files(date, self.path)    
+        
+        if not weather_files:
+            print('No Weather Date for these date.')
+            return
+
+        record_calculator = WeatherRecordsCalculation()
+        self.year_report = record_calculator.calculate_year_record(weather_files)
+
+        print (f'Highest: {self.year_report["highest_temp"]}C on {self.year_report["highest_temp_date"]}')
+        print(f'Lowest: {self.year_report["lowest_temp"]}C on {self.year_report["lowest_temp_date"]}')
+        print(f'Humidity: {self.year_report["humidity"]}% on {self.year_report["humidity_date"]}')
+    
+    def display_average_report_of_month(self, date: str) -> None:
+        """
+        Used for priniting report
+        
+        @params
+        :path str: Contains path to weather files directory.
+        :date str: Date entered by user as command line argument.
+        """
+        weather_file = WeatherRecordsReader.weather_files(date, self.path)    
```

Comment:
weather_file = WeatherRecordsReader.weather_files(date, self.path)
Since this thing is common in all methods, can we move it to the class level?